sdk/cmake/config.cmake

```diff
@@ -36,6 +36,20 @@ else()
 "Whether to compile for debugging.")
 endif()
 
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    set(GCC TRUE CACHE BOOL "The compiler is GCC")
+    set(CLANG FALSE CACHE BOOL "The compiler is LLVM Clang")
```

Comment:
We also have ```USE_CLANG_CL``` We should probably make this consistent. Not sure about the difference between CLANG / CLANG_CL and whether we support non-cl type of clang.

Reply:
This is our own option:
`option(USE_CLANG_CL "use clang-cl for msvc abi, or clang for gnu abi, same as clang --driver-mode=cl/gnu" ON)`